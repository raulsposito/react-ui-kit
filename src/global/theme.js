// https://www.color-hex.com/color/8293ee

export const colors = {
  primary: {
    regular: '#FC760D',
    dark: '#E46B0B',
    darker: '#C66F2B',
    hover: '#E17622',
    focus: '#B97641',
    selected: '#D4762C',
    disabled: '#F99546'
  },
  variant: {
    regular: '#FCB10D',
    dark: '#E29F0C',
    darker: '#C9A557',
    hover: '#E4A418',
    focus: '#BA923B',
    selected: '#D69E23',
    disabled: '#FBBF3C'
  },
  secondary: {
    regular: '#2FCFCF',
    dark: '#1F9A9A',
    darker: '#147676',
    hover: '#3DBDBD',
    focus: '#52A2A2',
    selected: '#44B4B4',
    disabled: '#5FD8D8'
  },
  success: {
    regular: '#3DB265',
    dark: '#48A668',
    darker: '#59946D',
    hover: '#48A668',
    focus: '#59946D',
    selected: '#59946D',
    disabled: '#9EBCA8'
  },
  error: {
    regular: '#FF5668',
    dark: '#E45C6B',
    darker: '#BA666F',
    hover: '#E45C6B',
    focus: '#BA666F',
    selected: '#BA666F',
    disabled: '#F8A4AC'
  },
  warning: {
    regular: '#FFAF00',
    dark: '#E5BB4D',
    darker: '#CCA644',
    hover: '#E4A418',
    focus: '#BA923B',
    selected: '#D69E23',
    disabled: '#FBBF3C'
  },
  contrast: {
    regular: '#3C3C3C',
    disabled: '#E4E4E4'
  },
  system: {
    regular: '#BAB5B4',
    dark: '#CBC6C4',
    darker: '#D7D4D1',
    hover: '#ACA8A8',
    focus: '#989595',
    selected: '#A6A2A1',
    disabled: '#C8C4C3'
  },
  background: {
    primary: '#FFFFFF',
    regular: '#FFFFFF',
    hover: '#F1F1F1',
    disabled: '#F8F8F8'
  },
  events: {
    inprogress: '#3DB265',
    next: '#5678CF',
    finished: '#BAB5B4',
    suspended: '#FFD056',
    cancelled: '#FF5668',
    white: '#FFFFFF'
  }
}

export const spacing = {
  quarter: '2px',
  half: '4px',
  threeQuarters: '6px',
  one: '8px',
  oneAndAQuarter: '10px',
  oneAndAHalf: '12px',
  two: '16px',
  twoAndAQuarter: '18px',
  twoAndAHalf: '20px',
  three: '24px',
  threeAndAHalf: '28px',
  four: '32px',
  fourAndAHalf: '36px',
  five: '40px',
  fiveAndAHalf: '44px',
  six: '48px',
  sixAndAHalf: '52px',
  seven: '56px',
  sevenAndAHalf: '60px',
  eight: '64px',
  eightAndAHalf: '68px',
  nine: '72px',
  nineAndAHalf: '76px',
  ten: '80px',
  tenAndAHalf: '84px',
  eleven: '88px',
  elevenAndAHalf: '92px',
  twelve: '96px',
  twelveAndAHalf: '100px',
  thirteen: '104px',
  fourteen: '112x',
  fifteen: '120px'
}

export const page = {
  width: '1200px'
}
