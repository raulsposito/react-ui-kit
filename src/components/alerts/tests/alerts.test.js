import React from 'react'
import { render, screen, fireEvent } from '@testing-library/react'

import { colors } from '@global/theme'

import Alert from '..'

describe('<Alert />', () => {
  it('Should match snapshot', () => {
    const { asFragment } = render(<Alert />)
    expect(asFragment()).toMatchSnapshot()
  })

  it('Renders with the correct title', () => {
    render(<Alert title='spiderman' type='success' />)
    expect(screen.queryByText('spiderman')).toBeInTheDocument()
  })

  it('Renders with the correct text', () => {
    render(<Alert text='spiderman' type='success' />)
    expect(screen.queryByText('spiderman')).toBeInTheDocument()
  })

  it('Can have buttons', () => {
    render(<Alert type='success' actions />)
    expect(screen.queryByRole('button')).toBeInTheDocument()
  })

  it('Default renders with no buttons', () => {
    render(<Alert type='success' />)
    expect(screen.queryByRole('button')).toBe(null)
  })

  it('Renders with the correct button text when buttonText is passed', () => {
    render(<Alert type='success' actions buttonText='clickMe' />)
    const button = screen.getByText('clickMe')
    expect(button).toBeInTheDocument()
  })

  it('Renders with the correct default button text if no prop is passed', () => {
    render(<Alert type='success' actions />)
    const defaultButton = screen.getByText('x')
    expect(defaultButton).toBeInTheDocument()
  })

  it('Can handle onClick', () => {
    const handleClick = jest.fn()
    render(<Alert type='success' actions buttonText='clickMe' buttonOnClick={handleClick} />)
    const button = screen.getByRole('button')
    fireEvent.click(button)
    expect(handleClick).toHaveBeenCalledTimes(1)
  })

  describe('Style', () => {
    it('Can have success style', () => {
      const { container } = render(<Alert type='success' />)
      const childNodes = container.firstChild
      expect(childNodes).toBeInTheDocument()
      expect(childNodes).toHaveStyle({ backgroundColor: `${colors.success.regular}` })
    })

    it('Can have warning style', () => {
      const { container } = render(<Alert type='warning' />)
      const child = container.firstChild
      expect(child).toBeInTheDocument()
      expect(child).toHaveStyle({ backgroundColor: `${colors.warning.regular}` })
    })

    it('Can have error style', () => {
      const { container } = render(<Alert type='error' />)
      const child = container.firstChild
      expect(child).toBeInTheDocument()
      expect(child).toHaveStyle({ backgroundColor: `${colors.error.regular}` })
    })
  })
})
