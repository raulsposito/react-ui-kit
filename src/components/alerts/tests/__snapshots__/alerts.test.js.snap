// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Alert /> Should match snapshot 1`] = `<DocumentFragment />`;
